web/src/message_fetch.js

```diff
@@ -602,18 +597,8 @@ export function initialize(fetched_latest_message) {
         }, consts.catch_up_backfill_delay);
     }
 
-    let anchor;
-    if (initial_pointer !== undefined) {
-        // If we're doing a server-initiated reload, similar to a
-        // near: narrow query, we want to select a specific message.
-        anchor = initial_pointer;
-    } else {
-        // Otherwise, we should just use the first unread message in
-        // the user's unmuted history as our anchor.
-        anchor = "first_unread";
-    }
     load_messages({
-        anchor,
+        anchor: "first_unread",
```

Comment:
We should test that with this change, your scroll position in "all messages" is preserved across a server-initiated reload. And also that if you click from "All messages" into specific topic, and then get a server-initiated reload, and go back to "all messages" via the back button, your scroll position is preserved.

I think that both should work, if I understand the logic correctly.

Reply:
Reloads will take user to the `first_unread`, this is actually fixed in the last commit as we borrow the behaviour we have set for other narrows, so it would be better if we merged this with the last commit. I opened #29271 for the other prep commits, so I can rebase this once that gets merged.